### ISSUE TITLE
[3.3.5] Spell: Warrior - Charge/Intercept/Intervene doesn't reach target if CCed during run animation.

### ISSUE THREAD

**Issue by Fimbulveter** [TrinityCore/TrinityCore#18161]
[//]: # (***************************************************************)
[//]: # (** DON'T DELETE THIS TEMPLATE ELSE YOUR ISSUE WILL BE CLOSED **)
[//]: # (***************************************************************)

**Description:**
When charging a target, the warrior should reach the target even if you were CC-ed during the running animation. It might apply for other classes that have the same type of running animation, but I am most familiar with warriors.

Sources:
http://www.warcraftmovies.com/movieview.php?id=157707 11:58 
http://www.youtube.com/watch?v=gx9_yykxk3k&t=1m1s

**Current behaviour:** 
The warrior stops its running animation when CC-ed and therefore doesn't reach its target.

**Expected behaviour:** 
The warrior should end up at the target when CC-ed during the running animation. After reaching the target, they should be CC-ed for whatever is left of the CC.

**Steps to reproduce the problem:**
Have 1 warrior and 2 enemies (one with any form of CC).
1. Warrior charges one target. 
2. Enemy casts any form of CC on the warrior during the running animation.
3. Watch as the warrior doesn't reach its target.

**Branch(es):** 3.3.5 atleast.

**TC rev. hash/commit:**  

**TDB version:**  
335.61

**Operating system:**  
Windows 7

[//]: # (This template is for problem reports. For other types of report, edit it accordingly.)
[//]: # (If this is a crash report, include the crashlog with https://gist.github.com/)
[//]: # (For fixes containing C++, create a Pull Request.)

**Comment by tkrokli** [comment]
You have not stated your core revision hash. TDB is outdated, new version was released 12 days ago.

**Comment by Fimbulveter** [comment]
Sorry, I picked up the latest TDB and added the TC. rev I am on.

The problem is still there with the newer TDB.

**Comment by Persephone** [comment]
confirm

**Comment by tkrokli** [comment]
@Persephone : core rev. hash / TDB version?

**Comment by akrom23** [comment]
The second movie - warrior does not reach target - only apllied stun.

**Comment by Fimbulveter** [comment]
In the second video, the warrior ends up by the target (reaching the target), not where he gets stunned (at the start of the charge). Right now it's possible to get stunned right at the start of the charge animation and not move a single yard.

**Comment by akrom23** [comment]
@Fimbulveter: no. Warrior applied stun immediately, then start charging. Stun stopped him in middle of road. As you can see, after stun passed, he walked a few steps to reach the target. This "visual" on the beggining of stun is caused by sent packets.

**Comment by Fimbulveter** [comment]
https://vid.me/ku8s (ignore the music, it was running in the background...) shows how it currently is. It's often more problematic when it's an initial stun like in the second video, you won't move a single yard then.

**Comment by akrom23** [comment]
For me it's working like in the second video - i don't see any difference.

**Comment by Fimbulveter** [comment]
@akrom23 Thank you, had to change this one too: https://github.com/TrinityCore/TrinityCore/blob/master/src/server/game/Entities/Unit/Unit.cpp#L13451

From my experience on playing on retail a few years back, I did never experience my charge spells failing like it has done recently here. I would say it's fairly easy to stop the running animation now, so if it was like this on Retail a lot of PvP guides would encourage players to stop this animation, but I can't really find anything on this. Whenever a warrior have been playing in any of Blizzard's World Championships (lan event), there has never been anyone trying to prevent the warrior from reaching the target, which makes me believe that the warrior should reach the target even if CC-ed.

**Comment by mikex2k** [comment]
Yes this is still the issue on the latest rev. A warrior can be cc'ed during charge and stuck where he was cc'ed mid-charge. cc's include scatter shot, pin/ravage, etc
